Algorithm.Python/AccumulativeInsightPortfolioConstructionModel.py

```diff
@@ -0,0 +1,59 @@
+﻿# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("System")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Algorithm.Framework")
+AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Orders import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Algorithm.Framework import *
+from QuantConnect.Algorithm.Framework.Alphas import *
+from QuantConnect.Algorithm.Framework.Execution import *
+from QuantConnect.Algorithm.Framework.Portfolio import *
+from QuantConnect.Algorithm.Framework.Risk import *
+from QuantConnect.Algorithm.Framework.Selection import *
+from datetime import timedelta
+
+### <summary>
+### Test algorithm using 'AccumulativeInsightPortfolioConstructionModel.py' and 'ConstantAlphaModel'
+### generating a constant 'Insight' 
+### </summary>
+class ConfidenceWeightedFrameworkAlgorithm(QCAlgorithm):
```

Comment:
Thanks for adding an algorithm!
I think it would be nice for both file and algorithm name to be aligned, maybe something like `AccumulativeInsightPortfolioRegressionAlgorithm`.
Could we add the C# equivalent, I see it was present in the previous PR https://github.com/QuantConnect/Lean/pull/3751/files#diff-de9ca8289336617ab4798e31de371a72R30. 
Note that the implementation of the algorithm should be adjusted to assert on the behavior of this new portfolio construction model, current implementation was designed to test weighted based portfolio construction models. Same as for the expected statistics https://github.com/QuantConnect/Lean/pull/3751/files#diff-de9ca8289336617ab4798e31de371a72R77
*It seems this algorithm file is pending to be added to the project

Reply:
Sorry, this was one aspect that I was still unsure of, so it was put as a placeholder until we could get proper regression tests. I can look into it now.

Reply:
Is there a way to run these tests locally before pushing the commits up? I'm on Linux...